Add IQS014 rule (#2378)
home-assistant/developers.home-assistant: docs/core/integration-quality-scale/rules/iqs014.md

---
title: "IQS014 - Entities use device classes where possible"
---

IQS014 is as follows:

Entities use device classes where possible

## Reasoning

Device classes are a way to give context to an entity.

> These are used by Home Assistant for various purposes like:

Reviewer: They are also used to adjust the representation of the entity in the Home Assistant UI.

> - Allowing the user to switch between different units of measurement than the device provides.

Reviewer 2: ```suggestion
- Allowing the user to switch to another unit of measurement than what the device provides.
```

- They are used for voice control to ask questions like "What is the temperature in the living room?".
- They are used for exposing entities to cloud based ecosystems like Google Assistant and Amazon Alexa.
- They are used to automatically set up a default set of icons for the entity.

> - They can be used for the name of the entity to decrease pressure on our translators.

Reviewer 2: ```suggestion
- They can be used to set a default name of the entity to decrease the burden on our translators.
```

Because of these reasons, it is important to use device classes where possible.

## Example implementation

In the example below we have a temperature sensor that uses the device class `temperature`.

> The name of this entity will be `My device temperature`.

Reviewer: Should we show an example where the translation is automatically provided by the device class?

`sensor.py`
```python
class MyTemperatureSensor(SensorEntity):
    """Representation of a sensor."""

    _attr_has_entity_name = True
    _attr_device_class = SensorDeviceClass.TEMPERATURE

    def __init__(self, device: Device) -> None:
        """Initialize the sensor."""
        self._attr_device_info = DeviceInfo(
            identifiers={(DOMAIN, device.id)},
            name="My device",
        )
```

## Additional resources

> A list of available device classes can be found in the platform pages under the [entity](../../entity) page.

Reviewer 2: It's not platform pages. A platform is eg hue light or sonos media player.
```suggestion
A list of available device classes can be found in the entity pages under the [entity](../../entity) page.
```
